### PR TITLE
[new release] cohttp-lwt-jsoo, cohttp, cohttp-lwt, cohttp-lwt-unix, cohttp-top, cohttp-async and cohttp-mirage (2.5.5)

### PR DESCRIPTION
**Pull request by mseri**
CoHTTP implementation for the Js_of_ocaml JavaScript compiler

- Project page: <a href="https://github.com/mirage/ocaml-cohttp">https://github.com/mirage/ocaml-cohttp</a>
- Documentation: <a href="https://mirage.github.io/ocaml-cohttp/">https://mirage.github.io/ocaml-cohttp/</a>

##### CHANGES:

- `Cohttp_async.resolve_local_file`, `Cohttp_lwt.resolve_local_file` and `Cohttp_lwt_unix.resolve_file`
  are now the same code under the hood (`Cohttp.Path.resolve_local_file`). The old names
  have been preserved for compatibility, but will be marked as deprecated in the next release. This
  changes the behavior of `Cohttp_lwt_unix.resolve_file`: it now percent-decodes the paths and blocks
  escaping from the docroot correctly. This also fixes and tests the corner cases in these methods
  when the docroot is empty. (@ewanmellor mirage/ocaml-cohttp#755)

### CODE REVIEW

**Comment by mseri** [comment]
@hannesm @dinosaure is there an extra upper bound needed for the conduit failure? Do you know what it should be

**Comment by samoht** [comment]
@mseri: yes there is a missing constraint: https://github.com/mirage/ocaml-conduit/releases/tag/v2.3.0

File: packages/cohttp-mirage/cohttp-mirage.2.5.5/opam [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+Please see <https://mirage.io> for a self-hosted explanation
+and instructions on how to use this library."""
+depends: [
+  "ocaml" {>= "4.05.0"}
+  "dune" {>= "1.1.0"}
+  "mirage-flow" {>= "2.0.0"}
+  "mirage-channel" {>= "4.0.0"}
+  "conduit" {>= "2.0.2"}
+  "conduit-mirage" {>= "2.0.2" & < "3.0.0"}
```
Comment (samoht) [inline packages/cohttp-mirage/cohttp-mirage.2.5.5/opam:26]:
```suggestion
  "conduit-mirage" {>= "2.0.2" & < "2.3.0"}
```

Comment (mseri) [reply packages/cohttp-mirage/cohttp-mirage.2.5.5/opam]: I have also upstreamed it to the backports branch

**Comment by samoht** [comment]
LGTM

**Comment by mseri** [comment]
Thanks!
